torch/csrc/lazy/ts_backend/ops/c10d.cpp

```diff
@@ -0,0 +1,47 @@
+#include <torch/csrc/distributed/c10d/ProcessGroup.hpp>
+#include <torch/csrc/lazy/core/metrics.h>
+#include <torch/csrc/lazy/core/tensor.h>
+#include <torch/csrc/lazy/ts_backend/ops/broadcast.h>
+#include <torch/library.h>
+
+namespace torch {
+namespace lazy {
+
+c10::intrusive_ptr<c10d::ProcessGroup::Work> broadcast(at::TensorList tensors,
+    const c10::intrusive_ptr<c10d::ProcessGroup>& process_group, int64_t root_rank, int64_t root_tensor,
+    int64_t timeout) {
+  TORCH_LAZY_FN_COUNTER("lazy::");
+
+  // Currently LazyTensor IR cannot interpret the eager FunctionSchema, i.e.,
+  // c10::intrusive_ptr<ProcessGroup::Work> broadcast(at::TensorList tensors, const c10::intrusive_ptr<ProcessGroup>& process_group, int64_t root_rank, int64_t root_tensor, int64_t timeout).
+  // For two reasons:
+  // 1: Don't understand the ProcessGroup::Work return type. LazyTensor IR expects returns types to be all Tensors.
+  // 2: If we change the return types to be a TensorList, it still won't work given TensorList is not supported as well.
+  // So here, we restrict the input to be one tensor and overload another broadcast FunctionSchema, i.e.,
+  // at::Tensor broadcast_(const at::Tensor& tensor, const c10::intrusive_ptr<ProcessGroup>& process_group, int64_t root_rank, int64_t root_tensor, int64_t timeout)
+  // In the new schema, it basically makes the broadcast as an inplace op.
+  TORCH_CHECK(tensors.size() == 1lu, "Only one tensor input is supported for c10d::broadcast.")
```

Comment:
Here is why.

Reply:
Oh, it's almost like Jiewen from 3 days ago just answered the question i asked 3 seconds a go!

Reply:
(tensorlist really doesn't work though?)

Reply:
I don't know for sure. I just didn't see that we support any at ops that return a TensorList. @JackCaoG Do you know?